File: app/src/main/java/com/example/aidlsampleapp/model/Student.java
package com.example.aidlsampleapp.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

/**
 * Created by Ajeena on 24-10-2019.
 * Claysol MediaLabs Pvt Ltd
 * ajeena.sainudeen@claysol.com
 */
public class Student implements Parcelable, Serializable {
    protected Student(Parcel in) {
    }

    public static final Creator<Student> CREATOR = new Creator<Student>() {
        @Override
        public Student createFromParcel(Parcel in) {
            return new Student(in);
        }

        @Override
        public Student[] newArray(int size) {
            return new Student[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
    }
}

File: app/src/main/java/com/example/aidlsampleapp/model/Student.kt
package com.example.aidlsampleapp.model

/**
 * Created by Ajeena on 28-11-2019.
 * Claysol MediaLabs Pvt Ltd
 * ajeena.sainudeen@claysol.com
 *
 */
class Student {
    var first_name: String = "Guest"
    var last_name: String = "Ghost"

    constructor(_first_name: String, _last_name: String) {// Secondary Constructor
        this.first_name = _first_name
        this.last_name = _last_name

    }


    class Student(_first_name: String, _last_name: String) {
        // Primary constructor
        var first_name  = _first_name
           /* get() = field
            set(value) {
                field = value
            }*/

        var last_name = _last_name
            get() = field
            set(value) {
                field = value
            }


    }


}